### ISSUE TITLE
Redis Sentinel error when Redis hosts use password

### ISSUE THREAD

**Issue by pgs-dskora** [ServiceStack/Issues#423]
This test code works fine when redis hosts don't use password but when password is required the following exceptions are thrown when redis master comes down:

Error No master found in: 127.0.0.1:6379, 127.0.0.1:6380
Error NOGOODSLAVE No suitable slave to promote

```
            var sentinelHosts = new[] { "127.0.0.1:5000", "127.0.0.1:5001", "127.0.0.1:5002" };
            var sentinel = new RedisSentinel(sentinelHosts, masterName: "mymaster");
            sentinel.HostFilter = host => "password@{0}".Fmt(host);
            var manager = sentinel.Start();

            sentinel.OnWorkerError = exception => Console.WriteLine(exception);

            while (true)
            {
                try
                {
                    const string RedisKey = "my Name";
                    using (var client = manager.GetClient())
                    {
                        var result = client.Get<string>(RedisKey);
                        Console.WriteLine("Redis Key: {0} \t Port: {1}", result, client.Port);
                    }
                }
                catch (Exception ex)
                {
                    Console.WriteLine($"Error {ex.Message}");
                }
                Thread.Sleep(3000);
            }
```

I use ServiceStack.Redis 4.0.55

**Comment by mythz** [comment]
How long ago did you install v4.0.55? If it was [more than 6 days ago](https://forums.servicestack.net/t/redis-noauth-authentication-required/2253/2?u=mythz) can you retry with the latest version from MyGet again after deleting the NuGet packages cache.

**Comment by pgs-dskora** [comment]
I installed it again today after deleting the NuGet packages cache.
The problem still exists.

**Comment by mythz** [comment]
Does the failover work without using ServiceStack.Redis, i.e. when you kill the master process does the SENTINEL server detect it and promote one of the slaves to master successfully?

If so can you please provide the redis.conf you're using for your redis master, slaves and sentinel also the version of Redis you're using and OS you're using it on, thx.

**Comment by pgs-dskora** [comment]
Failover on Redis master works as designed and sentinels promote the slave to master etc. 
When I disabled password in redis conifg and comment out this line of code: 

` sentinel.HostFilter = host => "password@{0}".Fmt(host);`

Everything works perfect. But with password enabled sentinel and redis works properly but provided code can't handel that. 
I use redis in version: 3.0.501 and working on Windows 10 x64

[redis.master.txt](https://github.com/ServiceStack/Issues/files/220442/redis.master.txt)
[redis.slave.txt](https://github.com/ServiceStack/Issues/files/220445/redis.slave.txt)
[sentinel1.txt](https://github.com/ServiceStack/Issues/files/220444/sentinel1.txt)
[sentinel2.txt](https://github.com/ServiceStack/Issues/files/220446/sentinel2.txt)
[sentinel3.txt](https://github.com/ServiceStack/Issues/files/220443/sentinel3.txt)

**Comment by xplicit** [comment]
@pgs-dskora Here is misconfiguration issue. According to redis documentation master and slave should have the same password, while in configs provider master redis server has password, while slave has not. Here is the link  http://redis.io/topics/sentinel#sentinel-and-redis-authentication

What I changed in your config to make it work properly. In `redis.master.config` I added the line:

     masterauth "password"

In `redis.slave.config` I added the line:
   
    requirepass "password"

After this was done switching to slave started to work without throwing exceptions.
